pkg/token/webbrowser.go

```diff
@@ -0,0 +1,76 @@
+package token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+type WebBrowserToken struct {
+	clientID    string
+	resourceID  string
+	tenantID    string
+	oAuthConfig adal.OAuthConfig
+}
+
+// newAzureCLIToken returns a TokenProvider that will fetch a token for the user currently logged into the Azure CLI.
+// Required arguments include an oAuthConfiguration object and the resourceID (which is used as the scope)
+func newWebBrowserTokenProvider(oAuthConfig adal.OAuthConfig, clientID, resourceID, tenantID string) (TokenProvider, error) {
+	if clientID == "" {
+		return nil, errors.New("clientID cannot be empty")
+	}
+	if resourceID == "" {
+		return nil, errors.New("resourceID cannot be empty")
+	}
+	if tenantID == "" {
+		return nil, errors.New("tenantID cannot be empty")
+	}
+
+	return &WebBrowserToken{
+		clientID:    clientID,
+		resourceID:  resourceID,
+		tenantID:    tenantID,
+		oAuthConfig: oAuthConfig,
+	}, nil
+}
+
+// Token fetches an azcore.AccessToken from the Azure CLI SDK and converts it to an adal.Token for use with kubelogin.
+func (p *WebBrowserToken) Token() (adal.Token, error) {
+	emptyToken := adal.Token{}
+
+	// Request a new Azure CLI token provider
+	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
+		TenantID: p.tenantID,
+		ClientID: p.clientID,
+	})
+	if err != nil {
+		return emptyToken, fmt.Errorf("unable to create credential. Received: %v", err)
+	}
+
+	// Use the token provider to get a new token
+	WebBrowserToken, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{p.resourceID}})
```

Comment:
local variable shouldn't start with upper case

Reply:
I found the bug: the resourceID needs to have `/.default` suffix. refer to https://github.com/Azure/azure-sdk-for-go/blob/c5358ab6c0bf824403f0ecdbd164c882d794793d/sdk/azidentity/MIGRATION.md#azidentity-5

Reply:
Addressed these two concerns in next commit.